pkg/recorder/diskrecorder/diskrecorder.go

```diff
@@ -102,17 +102,21 @@ func (d *DiskRecorder) SaveAtPath(records record.MemoryRecords, path string) (re
 
 // Prune the archives older than given time
 func (d *DiskRecorder) Prune(olderThan time.Time) error {
-	files, err := ioutil.ReadDir(d.basePath)
+	files, err := os.ReadDir(d.basePath)
 	if err != nil {
 		return err
 	}
 	count := 0
 	var errors []string
 	for _, file := range files {
-		if isNotArchiveFile(file) {
+		fileInfo, err := file.Info()
+		if err != nil {
+			errors = append(errors, err.Error())
```

Comment:
Don't we want to rather log an error and continue when the file doesn't exist? I guess `fileInfo` will be `nil` if the file doesn't exist and it will probably cause a panic later on or is that wrong assumption?

Reply:
Let me think...